app/models/secret_storage.rb

```diff
@@ -50,6 +50,43 @@ def self.keys
     end
   end
 
+  class HashicorpVault
+
+    SECRET_BACKEND ='secret/'.freeze
+    # we don't really want other directories in here,
+    # and there may be other chars that we find we don't like
+    ENCODINGS = {"/": "%2F"}
+
+    def self.read(key)
+      result = Vault.logical.read(SECRET_BACKEND  + self.encode(key))
+      result.data[:secret_data]
```

Comment:
what other metadata is in there ? ... would it make sense to show it in the UI -> return all ?